taskfiles/boost/tests.yaml

```
@@ -0,0 +1,74 @@
+version: "3"
+
+includes:
+  utils: "../../exports/taskfiles/utils/utils.yaml"
+
+vars:
+  G_BUILD_DIR: "{{.TASKFILE_DIR}}/build"
+  G_DEPS_DIR: "{{.G_BUILD_DIR}}/deps"
+  G_DEPS_CMAKE_SETTINGS_DIR: "{{.G_DEPS_DIR}}/cmake-settings"
+  G_DEPS_CMAKE_SETTINGS_FILE: "{{.G_DEPS_CMAKE_SETTINGS_DIR}}/settings.cmake"
+
+  G_BUILD_BOOST_TEST_DIR: "{{.G_BUILD_DIR}}/boost-test"
+  G_BOOST_TEST_CMAKE_CACHE: "{{.G_BUILD_BOOST_TEST_DIR}}/CMakeCache.txt"
+  G_BOOST_TEST_COMPILE_COMMANDS_DB: "{{.G_BUILD_BOOST_TEST_DIR}}/compile_commands.json"
+  G_BOOST_TEST_EXECUTABLE: "{{.G_BUILD_BOOST_TEST_DIR}}/boost-test"
```

1. Move the build directory to the root and not inline with source.
2. `G_DEPS_CMAKE_SETTINGS_FILE`, `G_BOOST_TEST_CMAKE_CACHE`, and `G_BOOST_TEST_COMPILE_COMMANDS_DB` are all unused.
3. Rename the variables a bit to avoid potential future collisions as many of these are commonly used at the top most level.

If this suggestion is used the names need to be changed in other places as well.
```suggestion
  G_BOOST_BUILD_DIR: "{{.ROOT_DIR}}/build"
  G_BOOST_DEPS_DIR: "{{.G_BOOST_BUILD_DIR}}/deps"
  G_BOOST_DEPS_CMAKE_SETTINGS_DIR: "{{.G_BOOST_DEPS_DIR}}/cmake-settings"

  G_BOOST_TEST_BUILD_DIR: "{{.G_BOOST_BUILD_DIR}}/boost-test"
  G_BOOST_TEST_EXECUTABLE: "{{.G_BOOST_TEST_BUILD_DIR}}/boost-test"
```